### DISCUSSION TITLE
G28/G28.1 vs G10 L2 P28

### DISCUSSION THREAD

**Discussion by mvackel** [Paciente8159/uCNC#534] [Ideas]
The uCNC wiki says that G28.1 is not supported and G10 L2 P28 should be used instead. I'm confuse on how to use this.
Using G28.1 and G28 is simple: position the machine in the desired physical Zero position and issue a "G28.1" command. This is the new Zero position.
To return to that position, issue a "G28" command, or it's variants.

How should the G10 L2 P28 command be used? What would be the similar G28 command?

By the way, "G28.1" command returns an error on uCNC, but G28 works, but moves to the fixed machine 0,0,0 coordinates.

**Comment by Paciente8159** [comment]
µCNC tries as best as possible to make the parser compliant to the RS274NGC v3 as possible (the standard is in the docs directory).
Yes G28.1 is not implemented in the core code because it's not part of the standard but can be implemented via extension module.

G10 L2 P28 expects you to define the G28 homing position in absolute/machine coordinates.

So if you want your G28 to send the machine to coordinates X20 Y10 you send that as argument to the G10 command.

In theory making this:
```
G53 G0 X20 Y10
G28.1
```
or this
```
G10 L2 P28 X20 Y10
```

is the same

After this sending a G28 will move your machine to X20Y10 (again in absolute coordinates)

**Comment by Paciente8159** [comment]
I will make this available as an extension module. This is one needed modification #537 .
The rest will be implemented as a new module.

**Comment by Paciente8159** [comment]
This is the proposed [module](https://github.com/Paciente8159/uCNC-modules/pull/38). Still needs testing. Depends on #537.
